plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -50,39 +54,54 @@ func RenderEventXML(eventHandle EvtHandle, renderBuf []byte) ([]byte, error) {
 	return utf16ToUTF8Bytes(renderBuf, bufferUsed)
 }
 
-func CreateBookmark(channel string, recordID uint64) (h EvtHandle, err error) {
+func CreateBookmark(w WindowsEventAPI, channel string, recordID uint64) (h EvtHandle, err error) {
 	xml := fmt.Sprintf(bookmarkTemplate, channel, recordID)
 	p, err := syscall.UTF16PtrFromString(xml)
 	if err != nil {
 		return 0, err
 	}
-	h, err = EvtCreateBookmark(p)
+	h, err = w.EvtCreateBookmark(p)
 	if err != nil {
 		return 0, fmt.Errorf("error when creating a bookmark. Details: %v", err)
 	}
 	return h, nil
 }
 
 func CreateQuery(path string, levels []string) (*uint16, error) {
-	var filterLevels string
-	for _, level := range levels {
-		if filterLevels == "" {
-			filterLevels = fmt.Sprintf(eventLogLevelFilter, level)
-		} else {
-			filterLevels = filterLevels + " or " + fmt.Sprintf(eventLogLevelFilter, level)
-		}
+	return createWindowsEventFilter(path, levels)
+}
+
+func CreateRangeQuery(path string, levels []string, r state.Range) (*uint16, error) {
+	rangeFilter := fmt.Sprintf(eventRangeFilter, r.StartOffset(), r.EndOffset())
+	return createWindowsEventFilter(path, levels, rangeFilter)
+}
+
+func createWindowsEventFilter(path string, levels []string, additionalFilters ...string) (*uint16, error) {
+	// Add log levels
+	var levelsFilter string
+	formattedLevels := make([]string, len(levels))
+	for i, level := range levels {
+		formattedLevels[i] = fmt.Sprintf(eventLogLevelFilter, level)
+	}
+	if len(formattedLevels) > 0 {
+		levelsFilter = fmt.Sprintf("(%s)", strings.Join(formattedLevels, " or "))
```

Comment:
nit: Could use a `string.Builder` instead of creating the slice and string. Not a big deal since this isn't called that frequently and there aren't that many levels/filters.